pages/home/script.js

```diff
@@ -377,19 +377,30 @@ const watchGeoLocationSuccess = async ({ coords }) => {
   const lng = coords?.longitude;
   geoMap.setMarkerOnMap(lat, lng);
 
-  // update current user position
-  position = {
-    lat,
-    lng,
-  };
-
   // If there is a report id in the query params and the focus param is set, don't pan to the user's location
   // but to the report's location
   if (flyToTrigger && !(!!idReport && (!!focusMarker || !!openDetail))) {
+    // update current user position
+    position = {
+      lat,
+      lng,
+    };
+
     await getReportApiCall(lat, lng);
     flyTo(lat, lng);
     flyToTrigger = false;
   }
+
+  const distanceDiff = geolocationDistance(lat, lng, position.lat, position.lng);
+
+  // if user moves more than 25Km's 
+  // change his current position to get new reports
+  if (distanceDiff > 25) {
+    position = {
+      lat,
+      lng,
+    };
+  }
 };
```

Comment:
If user's location has changed and has a difference of more than 25Km's considering we are fetching reports for a 50Km's  area.